src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormInputElementPartDisplayDriver.cs

```diff
@@ -35,10 +38,36 @@ public override async Task<IDisplayResult> UpdateAsync(FormInputElementPart part
         {
             context.Updater.ModelState.AddModelError(Prefix, nameof(viewModel.Name), S["A value is required for Name."]);
         }
+        else
+        {
+            var safeName = SanitizeInputName(viewModel.Name.AsSpan());
+
+            if (viewModel.Name != safeName)
+            {
+                context.Updater.ModelState.AddModelError(Prefix, nameof(viewModel.Name), S["A Name contains invalid characters."]);
+            }
+        }
 
-        part.Name = viewModel.Name?.Trim();
+        part.Name = viewModel.Name;
         part.ContentItem.DisplayText = part.Name;
 
         return Edit(part, context);
     }
+
+    public static string SanitizeInputName(ReadOnlySpan<char> inputSpan)
```

Comment:
We could move this into the `StringExtensions` to make it available for others. Additionally you could easily avoid allocations here if the string is already sanitized. But this is not performance critical I guess...

Reply:
@gvkries I moved it to StringExtensions based on your comment. If you are good with the last changes, please merge.